File: Task-2/task2_ws/src/turtle/src/sub_odom.cpp
#include <ros/ros.h>
#include <turtlesim/Pose.h>

using namespace ros;

void callback(const turtlesim::Pose::ConstPtr &pose)
{
    ROS_INFO("(x, y, theta) = (%f, %f, %f)\n", pose->x, pose->y, pose->theta);
    ROS_INFO("linear_velocity = %f\n", pose->linear_velocity);
    ROS_INFO("angular_velocity = %f\n", pose->angular_velocity);
}

int main(int argc, char **argv)
{
    // node
    init(argc, argv, "sub_odom", init_options::AnonymousName);
    NodeHandle nh;

    // subscriber
    Subscriber sub = nh.subscribe<turtlesim::Pose>("turtle1/pose", 1, callback);
    spin();

    return 0;
}

File: Task-2/task2_ws/src/turtle_pkg/scripts/sub_odom.py
#!/usr/bin/python3

import rospy
from turtlesim.msg import Pose


def callback(pose):
    rospy.loginfo("(x, y, theta) = (%f, %f, %f)\n", pose.x, pose.y, pose.theta)
    rospy.loginfo("linear_velocity = %f\n", pose.linear_velocity)
    rospy.loginfo("angular_velocity = %f\n",pose.angular_velocity)


def sub_odom():

    # node
    rospy.init_node("sub_odom", anonymous=True)

    # subscriber
    sub = rospy.Subscriber("turtle1/pose", Pose, callback)
    rospy.spin()

if __name__ == "__main__":
    try:
        sub_odom()
    except rospy.ROSInterruptException:
        pass
